File: javaArchive/src/Debt_LRA_Transcript/Transcript.java
package Debt_LRA_Transcript;

import java.util.List;
import java.util.ArrayList;

import lecture.Semester;
import person.Student;

//Sena VATANSEVER 150119755

public class Transcript {

	private Student student;
	private List<Semester> listOfSemester;
	private double gano;
	private int totalCreditsTaken;
	private int totalCreditsCompleted;
	private double points;

	public Transcript(Student student, List<Semester> listOfSemester) {
		super();
		this.student = student;
		this.listOfSemester = listOfSemester;
		if (this.listOfSemester == null) {
			this.listOfSemester = new ArrayList<Semester>();
		}
		this.totalCreditsTaken = totalCreditsTakenCalculator(this.listOfSemester);
		this.totalCreditsCompleted = totalCreditsCompletedCalculator(this.listOfSemester);
		this.points = pointsCalculator(this.listOfSemester);
		this.gano = points / totalCreditsTaken;
		
	}

	// Creating get and set methods for variables
	public Student getStudent() {
		return student;
	}
	public void setStudent(Student student) {
		this.student = student;
	}

	public List<Semester> getListOfSemester() {
		return listOfSemester;
	}

	public double getGano() {
		return gano;
	}

	public int getTotalCreditsTaken() {
		return totalCreditsTaken;
	}

	public int getTotalCreditsCompleted() {
		return totalCreditsCompleted;
	}

	public double getPoints() {
		return points;
	}

	public void addSemester(Semester semester) {
		listOfSemester.add(semester);
	}
	public void setListOfSemester(List<Semester> listOfSemester) {
		this.listOfSemester = listOfSemester;
	}
	public void setGano(double gano) {
		this.gano = gano;
	}
	public void setTotalCreditsTaken(int totalCreditsTaken) {
		this.totalCreditsTaken = totalCreditsTaken;
	}
	public void setTotalCreditsCompleted(int totalCreditsCompleted) {
		this.totalCreditsCompleted = totalCreditsCompleted;
	}
	public void setPoints(double points) {
		this.points = points;
	}
	
	private int totalCreditsTakenCalculator(List<Semester> listOfSemester) {
		int totalCreditsTaken = 0;
		for(Semester semester : listOfSemester) {
			totalCreditsTaken = totalCreditsTaken + semester.getCreditsTaken();
		}
		return totalCreditsTaken;
	}
	
	private int totalCreditsCompletedCalculator(List<Semester> listOfSemester) {
		int totalCreditsCompleted = 0;
		for(Semester semester : listOfSemester) {
			totalCreditsCompleted = totalCreditsCompleted + semester.getCreditsCompleted();
		}
		return totalCreditsCompleted;
	}
	
	private double pointsCalculator(List<Semester> listOfSemester) {
		double points = 0;
		for(Semester semester : listOfSemester) {
			points = points + semester.getPoints();
		}
		return points;
	}
}

File: pythonCode/Transcript.py
class Transcript:

    def __init__(self):
        pass
        
     # Creating properties for variables 
    
    def getStudent(self):
        return self.__student

    def setStudent(self,  student):
        self.__student = student
    
    def getListOfSemester(self):
        try:
            x = self.__listOfSemester
        except AttributeError:
            self.__listOfSemester = list()           
        return self.__listOfSemester

    def setListOfSemester(self, listOfSemester : list):
        self.__listOfSemester = listOfSemester

    def getGano(self):
        gano = self.__points / self.__totalCreditsTaken
        return self.__gano

    def setGano(self, gano : float):
        self.__gano = gano   

    def getTotalCreditsTaken(self):
        totalCreditsTaken = self.totalCreditsTakenCalculator(self.__listOfSemester)
        return self.__totalCreditsTaken

    def setTotalCreditsTaken(self, totalCreditsTaken : int):
        self.__totalCreditsTaken = totalCreditsTaken

    def getPoints(self):
        points = self.pointsCalculator()
        return self.__points

    def setPoints(self, points : float):
        self.__points = points

    def getTotalCreditsCompleted(self):
        totalCreditsCompleted = self.totalCreditsCompletedCalculator(self.__listOfSemester)
        return self.__totalCreditsCompleted

    def setTotalCreditsCompleted(self, totalCreditsCompleted : int):
        self.__totalCreditsCompleted = totalCreditsCompleted

    def getLastSemester(self):
        from Semester import Semester
        lastSemester = Semester()
        for s in range(0, len(self.__listOfSemester)):
            if s == (len(self.__listOfSemester)-1):
                lastSemester = s
        return self.getListOfSemester()[lastSemester - 1]

    # Creating other methods

    def addSemester(self, semester):
        self.__listOfSemester.append(semester)

    def totalCreditsTakenCalculator(self, listOfSemester : list):
        totalCreditsTaken = 0
        for semester in listOfSemester:
            totalCreditsTaken += semester.creditsTaken()
        return totalCreditsTaken

    def totalCreditsCompletedCalculator(self, listOfSemester : list):
        totalCreditsCompleted = 0
        for semester in self.__listOfSemester:
            totalCreditsCompleted += semester.creditsCompleted
        return totalCreditsCompleted

    def pointsCalculator(self):
        points = 0
        for semester in self.__listOfSemester:
            points += semester.points
        return points
